sticker-graph/sticker_graph/model.py

```diff
@@ -188,6 +200,23 @@ def subword_reprs(self):
         return tf.reshape(subword_reprs, [bytes_shape[0], bytes_shape[1],
                                           subword_reprs.shape[-1]])
 
+    def word_len(self, word, max_subword_len):
+        return tf.shape((tf.decode_raw(word, tf.uint8)[:max_subword_len]))[0]
+
+    def word_lens(self, words, max_subword_len):
+        shape = tf.shape(words)
+
+        # Flatten, because map_fn only works on axis 0
+        flat = tf.reshape(words, [-1])
+
+        # Decode words.
+        decoded = tf.map_fn(lambda x: self.word_len(x, max_subword_len), flat, dtype=tf.int32)
```

Comment:
`lengths` would be a more descriptive name

Reply:
unsure about the performance difference, but using something like `tf.count_nonzero` along the word axis could also return the word lengths without decoding and looping twice.

Reply:
I considered doing this, especially because the decoding is quite slow for some reason (didn't do any profiling). But I decided against it, since files are permitted to contain the NULL codepoint. It's bad for various reasons (e.g. because NULL is the terminator in C strings), but in the real-world files contain all kind of weird stuff. Of course, in such cases `count_nonzero` would give incorrect lengths.

One argument in favor of doing `count_nonzero` is that if tokens contain NULL all is probably lost anyway.

Reply:
The performance is the same, since the count_nonzero approach is shorter I have changed to that. I have thought a bit more about it, and I think that tokens will `NULL` characters are junk anyway, so it probably doesn't matter if we cut one or more characters by not counting `NULL`s.